interpreter/cling/lib/Interpreter/DefinitionShadower.cpp

```diff
@@ -57,9 +69,6 @@ namespace cling {
   }
 
   void DefinitionShadower::hideDecl(clang::NamedDecl *D) const {
-    assert(isClingShadowNamespace(D->getDeclContext())
```

Comment:
Why can this go?

Reply:
It was there to ensure that we were only removing entries corresponding to previous user-defined entities (i.e., part of a `__cling_N5xxx` namespace).

The intended behavior now is to be able to shadow decls that are possibly on other namespaces, e.g. on the `std` namespace.

Reply:
I read this as "the decl that wants to hide previous declarations *must come from user input*". As such, this assert should still hold? I believe what you are referring to ("shadow decls that are possibly on other namespaces") is called `Prev` in the code below?

Reply:
This `hideDecl()` function removes an (old) entry from the TU lookup table so that unqualified name lookup is not ambiguous.  Therefore, this assertion reads as "the declaration to be hidden (removed from the lookup table) came from user input".

This is no longer the case; the new behavior is to be able to shadow declarations no matter where the first definition came from (e.g. from a system header, such as `std::data`).